xstream/src/java/com/thoughtworks/xstream/converters/reflection/AbstractAttributedCharacterIteratorAttributeConverter.java

```diff
@@ -34,26 +35,9 @@
 public class AbstractAttributedCharacterIteratorAttributeConverter extends
     AbstractSingleValueConverter {
 
-    private static final Map instanceMaps = new HashMap();
-    private static final Method getName;
-    static {
-        Method method = null;
-        try {
-            method = AttributedCharacterIterator.Attribute.class.getDeclaredMethod(
-                "getName", (Class[])null);
-            if (!method.isAccessible()) {
-                method.setAccessible(true);
-            }
-        } catch (SecurityException e) {
-            // ignore for now
-        } catch (NoSuchMethodException e) {
-            // ignore for now
-        }
-        getName = method;
-    }
+    private static final Map instanceMaps = Collections.synchronizedMap(new HashMap());
```

Comment:
Semantically, a `synchronized` map is (of course) not the same as `ConcurrentHashMap` with `.computeIfAbesent()` (like in `master`) but it should be sufficient to prevent illegal concurrent access.
The drop in efficiency should be tolerable.